File: docs/oauth2-oidc/get-started/_common/code-examples/go/handlers/protected.go
// Profile handler (protected route)
func handleProfile(w http.ResponseWriter, r *http.Request) {
	// Get session cookie
	cookie, err := r.Cookie("session_id")
	if err != nil {
		http.Error(w, "No session found", http.StatusUnauthorized)
		return
	}

	// Get session from store
	session, ok := sessions[cookie.Value]
	if !ok {
		http.Error(w, "Invalid session", http.StatusUnauthorized)
		return
	}

	// Return profile data
	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(map[string]interface{}{
		"message": "This is protected data from the resource server",
		"user":    session.UserInfo,
	})
}

File: docs/oauth2-oidc/get-started/_common/code-examples/js/routes/protected.js
app.get("/profile", requireAuth, async (req, res) => {
  try {
    // Get user info
    const userInfo = await client.fetchUserInfo(
      config,
      req.session.tokens.access_token,
      client.skipSubjectCheck,
    )

    res.json({
      message: "This is protected data from the resource server",
      email: userInfo.email,
      sub: userInfo.sub,
    })
  } catch (error) {
    console.error("Profile error:", error)
    res.status(500).json({ error: `Error accessing profile: ${error.message}` })
  }
})
